metrics/storage_collector.go

```diff
@@ -0,0 +1,124 @@
+package metrics
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-evm-gateway/models"
+)
+
+type StorageSizeCollector struct {
+	//TODO: think of adding error metric indicating we couldn't update the db size
+	storageDir  string
+	storageSize prometheus.Gauge
+	interval    time.Duration
+	logger      zerolog.Logger
+	ctx         context.Context
+	cancel      context.CancelFunc
+	ready       chan struct{}
+}
+
+type StorageSizeCollectorOpts struct {
+	Factory    promauto.Factory
+	Logger     zerolog.Logger
+	StorageDir string
+	Interval   time.Duration
+}
+
+func NewRestartableStorageSizeCollector(opts StorageSizeCollectorOpts, retries uint) models.Engine {
+	engine := NewStorageSizeCollector(opts)
+	return models.NewRestartableEngine(engine, retries, opts.Logger)
+}
+
+func NewStorageSizeCollector(opts StorageSizeCollectorOpts) *StorageSizeCollector {
```

Comment:
just expose `NewStorageSizeCollector` we don't need to expose both (`NewRestartableStorageSizeCollector`). It's an implementation detail.